### PR TITLE
Update Readme.md, Login, Signup with email fixed, Custom theme and fonts feature introduce.

### PR DESCRIPTION
**Pull request by ajaman190**
### **Update readme, Firebase v9 new login and signup method.** #163

Hi team,
Thank you for taking the time to review my pull request for **GSOC23.**

**Summary:**
This pull request fix issues with Firebase v9 setup, login and sign up feature, dynamic page adjustment.

**Issues:**
1. Outdated Readme.md file instructions that did not match the new version setup instructions of Firebase v9, React Native CLI,  Google Map API, and Facebook authentication.
2. Login and signup with email and password were not functional.
3. All pages were static and not dynamically adjustable, as the size of components was styled with constant pixels.

**Checklists:**
- [x]  Updated the Readme.md file with the latest version setup instruction.
- [x]  Fixed the login and sign-up with email using the latest methods of `Firebase/Auth`.
- [x] Added a custom theme constants defined in `app/constants/themes.js`, which can dynamically resize the screen and its components according to the screen resolution.
- [ ]  Update all page styles with new dynamic theme method.

**Screen Recording:**

https://user-images.githubusercontent.com/84832565/224475476-2f4e1457-35e5-40b5-bb6f-2c8263c61423.mp4

I have made the necessary changes to the codebase. Please let me know if there are any further changes that need to be made before merging this pull request.

### CODE REVIEW

**Comment by shehand** [comment]
Few things to highlight. For each unique feature or bug fix, there should be associated issue raised in the issue section. And we should only raise a PR related to that issue. That is the standard procedure we usually follow.

So, seperate the changes here into different PRs. Ideally one PR should only address the readme update issue. And readme should ONLY cantains an abstract of what project is and basic setup guidelines. No need to add project contains.

When adding commits, please do not add your gh user id. Commit history contains all of that. Refer this -> [https://gist.github.com/shehand/72bc91ed5947283572eb78a393aeaf6b](https://gist.github.com/shehand/72bc91ed5947283572eb78a393aeaf6b)

In summary
1. Seperate this PR to unique one and each one should have an associated issue for it.
2. Try to follow some standard when committing your changes
